File: projects/10/java/Owl.java
/* Owl.java provides an Owl subclass of Bird.
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Completed by: Ian Christensen, for CS 214 at Calvin College
 * Date: Spring, 2018
 */

public class Owl extends Flying {

 /* explicit constructor
  * Receive: name, a String 
  * PostCond: myName == name.
  */
 public Owl() {
 	super();
 }

 public Owl(String name) {
 	super(name);
 }

 /* An Owl's Call
  * Return: an owl-call ("Whoo-hoo!").
  */
 public String call() {
 	return "Whoo-hoo!";
 }
} 

File: labs/10/ruby/Owl.rb
#!/usr/bin/ruby
# Owl.rb Defines the Owl class which inherits attributes and methods from the Bird superclass.
# Begun by: Dr. Adams, for CS 214 at Calvin College
# Completed by: Ian Christensen, for CS 214 at Calvin College
# Date: Spring, 2018

require './Bird.rb'

class Owl < Bird

	# Method: call, accessor for a Owl call
	# Precondition: Owl must exist
	# Receives: a Owl object
	# Returns: 'Whoo-hoo', a owl call
	def call
		'Whoo-hoo'
	end
end
